### DISCUSSION TITLE
Outreachy Discussions - 2023 Round 2

### DISCUSSION THREAD

**Discussion by MridulS** [networkx/networkx#6960] [General]
Hello all!

Please use this discussion page to discuss everything for Outreachy 2023 round 2 (December 2023 - March 2024).

If you have any questions or doubts about the processes please put in a reply to this discussion post.

As a first step: applicants who haven't used networkx before should first [install networkx](https://networkx.org/documentation/latest/install.html) on their machines and follow our [introductory tutorial](https://networkx.org/documentation/latest/tutorial.html). This will give applicants a better overview of our codebase and API, which will help the applicant create a strong application. If something isn't clear from the tutorial please raise an issue/put a comment here so we can do a better job of maintaining our tutorial.

Once applicants are comfortable with using networkx, they should go through our [Contributor Guide](https://networkx.org/documentation/latest/developer/contribute.html) to setup our local development environment for networkx.

We have created another repository https://github.com/networkx/outreachy which will have explicit tasks for Outreachy applicants during the contribution period.

To participate with NetworkX as an Outreachy intern applicants are expected to go through multiple tasks in that repository. All tasks are presented as [issues](https://github.com/networkx/outreachy/issues).

Everyone is expected to start with https://github.com/networkx/outreachy/issues/2.

The applicant should also go through the [New Contributor FAQs](https://networkx.org/documentation/latest/developer/new_contributor_faq.html) as it may answer some of their questions.

As the project will deal with creating new notebooks for our NX-guides please do go through current notebooks at https://networkx.org/nx-guides/.

As the project will deal with implementing new algorithms for nx-parallel, please go through github.com/networkx/nx-parallel/

If the applicant is interested in working on specific issues they can comment on them. Applicants can also pick any [open issue](https://github.com/networkx/networkx/issues) if they like or even open a new issue/discussion if there is something in our documentation or codebase that doesn't make sense :)

Feel free to post here if you have any other questions or issues!

Excited for the coming months! :)

**Comment by akshayamadhuri** [comment]
as for outreachy contribution do we need to contribute only to this repo https://github.com/networkx/outreachy or can we contribute to this https://github.com/networkx/networkx  also as outreachy contribution

**Reply by MridulS** [reply]
One of the the tasks in the outreachy repo is to contribute to https://github.com/networkx/networkx https://github.com/networkx/outreachy/issues/4

**Reply by akshayamadhuri** [reply]
@MridulS   hello, I'm new here,do we need to contribute to multiple issues or can we raise a single PR on this repo:https://github.com/networkx/outreachy

**Reply by MridulS** [reply]
Hello @akshayamadhuri, the initial expectation is to start with https://github.com/networkx/outreachy/issues/2 and then work on other issues in the outreachy repo. The issues https://github.com/networkx/outreachy/issues explain how and what you need to do to create a PR :)

**Comment by Ytemiloluwa** [comment]
Hello ! My name is Temi, I am an outreachy applicant willing to contribute to NetworkX.

**Reply by MridulS** [reply]
Welcome! Feel free to ask any questions if you have any doubts @Ytemiloluwa :)

**Comment by Olauryn** [comment]
Hello Everyone. My name is Lauryn and I'm interested in contributing to NetworkX Outreachy tasks

**Reply by MridulS** [reply]
Welcome! Feel free to ask any questions if you have any doubts @Olauryn :)

**Reply by Olauryn** [reply]
Thank you

**Comment by Adenike-star** [comment]
Hello everyone! I'm Adenike, an outreachy applicant. I'm interested in contributing to NetworkX project.

**Reply by MridulS** [reply]
Welcome! Feel free to ask any questions if you have any doubts @Adenike-star  :)

**Reply by Adenike-star** [reply]
Okay, thank you.

**Comment by khushishikhu** [comment]
Hi everyone, I am Khushi Gautam an outreachy applicant for december cohort, I am interested in working with this organization. Looking forward to connect with you all in this journey.

**Comment by ThandolwethuM** [comment]
Hello fellow coders, my name is Thando. I see I'm a bit late to the party but I'm more than ready to dive head first. Can't wait to collaborate with everyone!

**Comment by ThandolwethuM** [comment]
I would just like to confirm 2 things. I don't remember filling in the project name and mentor in my initial application but I read this under initial application at https://www.outreachy.org/apply/initial-application/. And I can't seem to find where I can apply for this project specifically. Could somebody please clarify or share an exact link where I can start

**Reply by MridulS** [reply]
I believe https://www.outreachy.org/outreachy-december-2023-internship-round/communities/networkx/#implement-new-parallel-algorithms-in-nx-parallel should take you to the project page.

**Reply by axif0** [reply]
Hello @ThandolwethuM , so for applying for this project you will need to click on [`record your contributions and create a final application`](https://www.outreachy.org/outreachy-december-2023-internship-round/communities/networkx/implement-new-parallel-algorithms-in-nx-parallel/contributions/) button in outreachy website and then in redirected page you will have to `Record a Contribution`. PRs you are working on is going to be counted as your contribution. 

Secondly, for this project we, the applicents are working on these issues - https://github.com/networkx/outreachy/issues

I hope it helps!

**Comment by khushishikhu** [comment]
I have a doubt while setting up the first step, in which python environment I need run 
`>>> import networkx as nx
>>> print(nx.__version__)
<note down this output>`
these commands. I tried on jupyter notebook and it failed

**Reply by akshayamadhuri** [reply]
@khushishikhu  try it on replit

**Comment by Bonheur773** [comment]
Good evening everybody, my name is Bonheur from Rwanda, even though I am a bit late but I am excited and ready to work with you guys on this project!!

**Comment by spykelionel** [comment]
Hello @MridulS and everyone, I am Spyke Lionel, an Outreachy applicant. I'm glad to be here.

**Comment by SiziweLupondo** [comment]
Good day everyone, i am Siziwe from South Africa. Nice to be here.

**Comment by Anila-Ali** [comment]
Hi everyone, I am Anila Ali and I am interested to contribute in this project.

**Comment by ThandolwethuM** [comment]
Hi, @everyone. Has anyone been able to successfully install NetworkX on their machines?

**Reply by MridulS** [reply]
Hi @ThandolwethuM What issue are you running into?

**Reply by ThandolwethuM** [reply]
I have installed it, but I get this error when I try to run the tutorial python script on Jupyter:
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[2], line 2
      1 import networkx as nx
----> 2 G = nx.Graph()

AttributeError: module 'networkx' has no attribute 'Graph'

**Reply by ThandolwethuM** [reply]
And if I download it as a python code file and run it, this is the error:
    import networkx as nx
ModuleNotFoundError: No module named 'networkx'.

**Reply by Schefflera-Arboricola** [reply]
@ThandolwethuM did you try running `!pip install networkx` in Jupyter or `pip install networkx` in the terminal?

**Reply by khushishikhu** [reply]
@ThandolwethuM Try running it in development environment (env) file

**Reply by ThandolwethuM** [reply]
@Schefflera-Arboricola Yes, I did. and it downloaded everything.... I'm thinking maybe the issue could be the location of my files.

**Reply by Schefflera-Arboricola** [reply]
@ThandolwethuM Could you pls tell me more about how and where you are running the scripts etc. so that I can help you better?

**Reply by Ytemiloluwa** [reply]
> Hi, @everyone. Has anyone been able to successfully install NetworkX on their machines?
> 

Hello @ThandolwethuM 
Start by checking which version of Python you have on your system. Networkx require python 3.8, 3.9 and 3.10 to install NetworkX on your machine. 
You can install NetworkX on your machine with this command. Run the command in your terminal/cmd

$ pip install networkx[default]

Here is also the installation instructions on the documentation website 

https://networkx.org/documentation/stable/install.html

I hope it helps 🙏

**Reply by Ytemiloluwa** [reply]
> @ThandolwethuM Try running it in development environment (env) file

@ThandolwethuM try running the script in the python IDLE environment 

You can start by searching for IDLE in your system

**Reply by axif0** [reply]
@ThandolwethuM you can also use online IDEs Google Colab if you still face problem, it will help you to unblock quickly if you are working on issue no5. 
Please let us know if you still face any further problem.

**Comment by maitreyamarg** [comment]
Hi everyone, maitreya from India. I'm looking forward to learning and contributing as much as I can! thanks for this space.

**Comment by Ytemiloluwa** [comment]
Hello @MridulS I will like to ask if there is a timeline for the project : Implement new parallel algorithms in nx parallel. If there is a timeline then on behalf of all applicants, I will like ask for it so we can include it in our final applications. Thank you.

**Reply by akshayamadhuri** [reply]
@Ytemiloluwa  i think its an internship task

**Reply by axif0** [reply]
As this project is still open, I guess we have time till 30 October.

**Reply by MridulS** [reply]
@Ytemiloluwa we would like for interns to come up with the timeline, feel free to share your timeline here if you want to get a review :)

**Reply by Ytemiloluwa** [reply]
@MridulS okay, that is awesome.

**Comment by tiwavaldese** [comment]
Hello everyone i'm Tiwa valdese and outreachy applicant, please i will like to contribute to this project

**Reply by tiwavaldese** [reply]
please can someone help, i'm stuck in activating the virtual env using source networkx-dev/bin/activate. this is what i got when i try to run in the terminal **source networkx-dev/bin/activate
bash: networkx-dev/bin/activate: No such file or directory**

**Reply by Schefflera-Arboricola** [reply]
@tiwavaldese did you run `python -m venv networkx-dev` before running `source networkx-dev/bin/activate` in the root dir?

**Reply by tiwavaldese** [reply]
yes i did

**Reply by spykelionel** [reply]
@tiwavaldese Kindly share screenshots instead

**Reply by tiwavaldese** [reply]
![2f](https://github.com/networkx/networkx/assets/119421647/922a8031-47ec-4124-a0b3-9cddd9b8fe32)

**Reply by spykelionel** [reply]
Do `ls` or `dir` let's see

**Reply by spykelionel** [reply]
I mean type the command `ls` or `dir` and send the ouput

**Reply by spykelionel** [reply]
`source network-dev/bin/activate` is for Linux/MacOS

You seem to be using Windows OS. In that case, do `network-dev\Scripts\activate` instead.

**Reply by tiwavaldese** [reply]
@spykelionel thanks alot this **networkx-dev/scripts/activate** work

**Reply by spykelionel** [reply]
You are welcome! I'm glad I could be of assistance.

**Comment by tiwavaldese** [comment]
Hello Mr @MridulS 
please i'm having issues to apply for the networkx project. i'm stuck at filling the **Community-specific Questions**,**outreachy internship project timeline** and in recording the contribution, will i put the **repo link or the deploy link**? thank you

**Reply by MridulS** [reply]
> Community-specific Questions

We don't have anything specific we need, but feel free to provide any details you think would be helpful :)

> outreachy internship project timeline

We expect you to provide a rough timeline about how you plan on working on the project. Feel free to share your timeline if you would like to get a review before submitting the application :)

> repo link or the deploy link

Please put in the links to your pull requests in networkx/outreachy and networkx/networkx repositories.

**Reply by tiwavaldese** [reply]
thank you for the clarification

**Comment by akshayamadhuri** [comment]
@MridulS can you tell me how to submit our final application do we need to make a doc with timeline

**Reply by MridulS** [reply]
We expect you to provide a rough timeline about how you plan on working on the project. Feel free to share your timeline if you would like to get a review before submitting the application :)

You can create a weekly/biweekly plan.

**Comment by akshayamadhuri** [comment]
@spykelionel I think we need to add a doubts clear sessions and meetups in between the weeks too and discussing about the test cases with various env

**Reply by spykelionel** [reply]
Hi, @akshayamadhuri! Thanks for pointing that out. :)
